packages/tools/client-debug-view/src/components/data-object-views/SharedCounterView.tsx

```diff
@@ -37,12 +53,66 @@ export function SharedCounterView(props: SharedCounterViewProps): React.ReactEle
 		};
 	}, [sharedCounter, setValue]);
 
+
+    function decrementCounter(): void {
+        sharedCounter.increment(-1);
+	}
+
+    function incrementCounter(): void {
+		sharedCounter.increment(1);
+	}
+
+    const inputSetValue = (e: any): void => {
+        numberFormatCheck(e.target.value);
+    }
+
+    const numberFormatCheck = (val: number): void => {
+        const num = val ?? value;
+
+        if (!isFinite(num) || num[0] === '-') {
+            return
+        }
+
+        sharedCounter.increment(num - value);
+    }
+
 	return (
 		<Stack>
 			<StackItem>
 				<b>SharedCounter</b>
 			</StackItem>
-			<StackItem>Value: {value}</StackItem>
+			<StackItem>Value: {value} </StackItem>
+            <StackItem>
+                <TooltipHost
```

Comment:
We want to be able to increment and decrement by arbitrary integer amounts here, not just by 1. I think we want to use the text field in the middle to let users type in an integer value delta, and use the plus/minus buttons to submit an increment/decrement of that amount to the shared counter. We'll also want to display the current value of the counter alongside it (or above it, as you had drawn in our whiteboarding session).

Reply:
Changed the logic with additional `React.useState` of `[deltaValue, setDeltaValue]`